### ISSUE TITLE
extension encode error

### ISSUE THREAD

**Issue by CoderHuo** [eerimoq/asn1tools#24]
I am a new pythoner to ues the asn1tools in python3.65 enviroment.
I found a mistake below.

``` python
foo = '''
Foo DEFINITIONS ::= BEGIN

    PosProtocol ::= SEQUENCE {
        tia801  BOOLEAN,
        rrlp    BOOLEAN,
        rrc     BOOLEAN,
        ...,
        ver2-PosProtocol-extension Ver2-PosProtocol-extension
    }

    Ver2-PosProtocol-extension ::= SEQUENCE {
        lpp     BOOLEAN,
        posProtocolVersionRRLP   PosProtocolVersion3GPP  OPTIONAL,
        ...,
        lppe          BOOLEAN OPTIONAL}

    PosProtocolVersion3GPP ::= SEQUENCE {
        majorVersionField      INTEGER(0..255),
        technicalVersionField  INTEGER(0..255),
        editorialVersionField  INTEGER(0..255),
        ...}
END
'''

testdata = {
    'tia801': False,
    'rrlp': True,
    'rrc': False,
    'lpp': False,
   #ver2-PosProtocol-extension: {
    'posProtocolVersionRRLP': {
        'majorVersionField': 7,
        'technicalVersionField': 11,
        'editorialVersionField': 0
    }
    #}
}
t1 = asn1tools.compile_string(foo, 'uper')
t1_encode = t1.encode('PosProtocol', testdata)
print(binascii.b2a_hex(t1_encode))

t2_decode = t1.decode('PosProtocol', t1_encode)
print(t2_decode)
```

The first output is    **b'a010480e160000'**
then I decode the t1_encode, but the decode result is wrong.
The second output is   **{'tia801': False, 'rrlp': True, 'rrc': False, 'lpp': False, 'lppe': False}** , not same as the testdata .
So I look the source code , may be the encode_additions function have same mistakes.
but it's difficult to me to find the root cause.

Thank you.

**Comment by CoderHuo** [comment]
The first output should be "A01044070B0000"

**Comment by eerimoq** [comment]
Fixed in version 0.80.1. There was a mistake in the code for encoding/deciding sequence extensions.

``` python
import asn1tools
import binascii

foo = '''
Foo DEFINITIONS ::= BEGIN

    PosProtocol ::= SEQUENCE {
        tia801  BOOLEAN,
        rrlp    BOOLEAN,
        rrc     BOOLEAN,
        ...,
        ver2-PosProtocol-extension Ver2-PosProtocol-extension
    }

    Ver2-PosProtocol-extension ::= SEQUENCE {
        lpp     BOOLEAN,
        posProtocolVersionRRLP   PosProtocolVersion3GPP  OPTIONAL,
        ...,
        lppe          BOOLEAN OPTIONAL}

    PosProtocolVersion3GPP ::= SEQUENCE {
        majorVersionField      INTEGER(0..255),
        technicalVersionField  INTEGER(0..255),
        editorialVersionField  INTEGER(0..255),
        ...}
END
'''

testdata = {
    'tia801': False,
    'rrlp': True,
    'rrc': False,
    'ver2-PosProtocol-extension': {
        'lpp': False,
        'posProtocolVersionRRLP': {
            'majorVersionField': 7,
            'technicalVersionField': 11,
            'editorialVersionField': 0
        }
    }
}

t1 = asn1tools.compile_string(foo, 'uper')
t1_encode = t1.encode('PosProtocol', testdata)
print(binascii.b2a_hex(t1_encode))

t2_decode = t1.decode('PosProtocol', t1_encode)
print(t2_decode)
```

The output:

```
> python3 test.py
b'a01044070b0000'
{'tia801': False, 'rrlp': True, 'rrc': False, 'ver2-PosProtocol-extension': {'posProtocolVersionRRLP': {'majorVersionField': 7, 'technicalVersionField': 11, 'editorialVersionField': 0}, 'lpp': False}}
> 
```

**Comment by CoderHuo** [comment]
Thank you very much.
